src/vec3.rs

```diff
@@ -1,324 +1,219 @@
-use crate::{random_f64, utils::random_f64_minmax};
+use rand::Rng;
 use std::fmt;
 use std::ops;
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, Default)]
 pub struct Vec3 {
     pub x: f64,
     pub y: f64,
     pub z: f64,
 }
 
-pub use Vec3 as Point3;
-pub use Vec3 as Color;
+pub type Point3 = Vec3;
+pub type Color = Vec3;
 
-impl Vec3 {
-    pub fn new() -> Vec3 {
-        Vec3 {
-            x: 0.,
-            y: 0.,
-            z: 0.,
+macro_rules! impl_binop {
```

Comment:
I really like this implementation! However, explaining the macros for the beginner would be slightly too much.

Reply:
Hmm, I though it might be a nice "hey, we have macros in rust" introduction, but I understand if you think it adds too much complexity...

Reply:
I mention it and the relevant changes in the updated text 😁

Reply:
And honestly, Rust macros are a potent beast, much more powerful than C ones imo

Reply:
Yeah, I think they fit in very well with rust, since they restrict you a bit more (than C), but most of the thinks you can't do are a bad idea anyways and in that they go very well with the whole pattern matching themw